rcall.c

```diff
@@ -764,10 +781,21 @@ static void pg_get_one_r(char *value,  plcDatatype column_type, SEXP *obj, int e
                            plc_get_type_name(column_type), column_type);
             break;
 
+        case PLC_DATA_BYTEA:
         case PLC_DATA_TEXT:
         default:
             /* Everything else is defaulted to string */
-            SET_STRING_ELT(*obj, elnum, COPY_TO_USER_STRING(((char *)value)));
+			if (strlen(value) != 0 && column_type == PLC_DATA_BYTEA){
+				/*for bytea type, we first get its size then do copy*/
+		    	bsize = *((int*)value);
+				temp = (char *)pmalloc(bsize + 1);
+				strncpy (temp, value + 4, bsize);
+				temp[bsize] = '\0';
+				SET_STRING_ELT(*obj, elnum, COPY_TO_USER_STRING(temp));
```

Comment:
If bytea allows '\0' before the end, then the code is buggy. Please check, either fix or disable the type.